test/dialect/mssql/test_compiler.py

```diff
@@ -1281,6 +1281,31 @@ def test_index_extra_include_2(self):
             schema.CreateIndex(idx), "CREATE INDEX foo ON test (x) INCLUDE (y)"
         )
 
+    def test_index_include_where(self):
+        metadata = MetaData()
+        tbl = Table(
+            "test",
+            metadata,
+            Column("x", Integer),
+            Column("y", Integer),
+            Column("z", Integer),
+        )
+        idx = Index(
+            "foo", tbl.c.x, mssql_include=[tbl.c.y], mssql_where=tbl.c.y > 1
+        )
+        self.assert_compile(
+            schema.CreateIndex(idx),
+            "CREATE INDEX foo ON test (x) INCLUDE (y) WHERE y > 1",
+        )
+
+        idx = Index(
+            "foo", tbl.c.x, mssql_include=[tbl.c.y], mssql_where="y > 1"
```

Comment:
@zzzeek - This test passes under 1.4 but fails under 1.3 with "AttributeError: 'str' object has no attribute '_compiler_dispatch'". Should we wrap this particular test in an `if` block to bypass it for versions < "1.4"?

Reply:
it looks like this test was added in 69502725db4829a84872697fd65696 and in fact there's no reason mssql_where="string" should even work...I would hope this is emitting a deprecation warning in fact.    For now I would use text("y > 1") as the argument which is the normal way to do this.

Reply:
Done. Thanks.